inflect/__init__.py

```diff
@@ -3201,6 +3209,14 @@ def _sinoun(  # noqa: C901
         words = Words(word)
 
         if words.last.lower() in pl_sb_uninflected_complete:
+            if len(words.split_) >= 3:
+                for numword in range(1, len(words.split_) - 1):
+                    if words.split_[numword] in pl_prep_list_da:
+                        return " ".join(
+                            words.split_[: numword - 1]
+                            + [self._sinoun(words.split_[numword - 1], 1)]
+                            + words.split_[numword:]
+                        )
```

Comment:
This looks like a copy/paste of other code. Although existing `inflect` logic may have used copy/paste as a way to re-use behavior, I'd like to avoid it for future contributions. Let's either first refactor the common behavior of `_sinoun` and `_plnoun` so that the change is only needed once or at the very least provide this functionality as its own method so that only one line needs to be repeated in both methods. I acknowledge that may prove difficult given the nested conditionals, loops, and inner return. Let me know if you'd like my help with that.

Reply:
Thanks for the help offering! I came up with this, so let me know what you think: https://github.com/jaraco/inflect/pull/188/commits/712906cc85bf0942fd540263965c104936b44f9d

Reply:
Much better. Thanks!